src/dbnode/storage/index.go

```diff
@@ -524,8 +539,15 @@ func (i *nsIndex) writeBatches(
 				batch.MarkUnmarkedEntryError(m3dberrors.ErrTooPast, idx)
 				return
 			}
+
+			if i.forwardIndexDice.forwardIndexing(entry.Timestamp) {
+				forwardWriteEntry := entry
+				forwardWriteEntry.Timestamp = entry.Timestamp.Add(blockSize)
```

Comment:
Is the timestamp used for anything other than figuring out which block to put it in? Also, I may be wrong but my understanding of the forward indexing logic seemed to imply that you could forward index stuff that wasn't yet within the bufferFuture window. Is that correct? If so, is it guaranteed there is a block available for this write to get indexed into?

Reply:
Given the current behaviour, it will only forward index within a percentage of the `bufferFuture` window